### PR TITLE
Docstring fixes for unchanged functions

### PR DESCRIPTION
**Pull request by mudit2812**
This PR adds some fixes to the documentation of functions that did not get updated for the v0.29 release.

~~Note: More changes are being added, so PR has been marked as draft for now.~~

### CODE REVIEW

File: pennylane/qchem/tapering.py [hunk 1]
```diff
@@ -182,8 +182,9 @@ def paulix_ops(generators, num_qubits):
     These are required to obtain the Clifford operators :math:`U` for the Hamiltonian :math:`H`.
 
     Args:
-        generators (list[Hamiltonian]): list of generators of symmetries, taus, for the Hamiltonian
+        generators (list[Hamiltonian]): list of generators of symmetries, :math:`tau`s, for the Hamiltonian
         num_qubits (int): number of wires required to define the Hamiltonian
+
```
Comment (rmoyard) [inline pennylane/qchem/tapering.py:187]: /github/workspace/pennylane/qchem/tapering.py:docstring of pennylane.qchem.tapering.paulix_ops:5: WARNING: Inline interpreted text or phrase reference start-string without end-string.

**Review by rmoyard — COMMENTED** [review]
It looks good expect the following issue.

File: pennylane/pauli/conversion.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def pauli_decompose(
 
     Returns:
         Union[~.Hamiltonian, ~.PauliSentence]: the matrix decomposed as a linear combination
-            of Pauli operators, either as a :class:``~.Hamiltonian`` or :class:``~.PauliSentence`` instance.
+        of Pauli operators, either as a :class:``~.Hamiltonian`` or :class:``~.PauliSentence`` instance.
```
Comment (albi3ro) [inline pennylane/pauli/conversion.py:46]:
```suggestion
        of Pauli operators, either as a :class:`~.Hamiltonian` or :class:`~.PauliSentence` instance.
```

File: pennylane/transforms/metric_tensor.py [hunk 3]
```diff
@@ -63,6 +63,7 @@ def metric_tensor(
     Args:
         tape (pennylane.QNode or .QuantumTape): quantum tape or QNode to find the metric tensor of
         argnum (int or Sequence[int] or None): Trainable tape-parameter indices with respect to which
+            the metric tensor is computed. If `argnum=None`, the metric tensor with respect to all
```
Comment (albi3ro) [inline pennylane/transforms/metric_tensor.py:66]:
```suggestion
```
? Looks like this line somehow got duplicated?

File: pennylane/transforms/optimization/pattern_matching.py [hunk 4]
```diff
@@ -91,9 +91,9 @@ def circuit():
     1: ──Z──╰Z─╭●─┤
     2: ──Z─────╰Z─┤
 
-    Note that with this pattern we also replace a ``pennylane.S``, ``pennylane.PauliZ`` sequence by ``pennylane.S``.
-    If one would like avoiding this, it possible to give a custom quantum cost dictionary with a negative cost for
-    ``pennylane.PauliZ``.
+    Note that with this pattern we also replace a ``pennylane.S``, ``pennylane.PauliZ`` sequence by
+    ``pennylane.Adjoint(pennylane.S)``. If one would like avoiding this, it possible to give a custom
```
Comment (albi3ro) [inline pennylane/transforms/optimization/pattern_matching.py:95]:
```suggestion
    ``Adjoint(S)``. If one would like avoiding this, it possible to give a custom
```
The `Adjoint` class is not available in the `pennylane` namespace, just `pennylane.ops.op_math`, as "adjoint" is the user facing constructor.  So maybe we should just write it out this way?

**Review by albi3ro — COMMENTED** [review]
Excellent find on some of these.  One or two minor comments.

File: pennylane/transforms/optimization/pattern_matching.py (hunk 4 above)
Comment (mudit2812) [reply pennylane/transforms/optimization/pattern_matching.py:95]: Good idea

**Review by albi3ro — APPROVED** [review]
👍 🎉

**Review by rmoyard — APPROVED** [review]
Thanks!

**Pull request merged**